pkg/exposer/csi_snapshot.go

```diff
@@ -0,0 +1,386 @@
+		}
+	}()
+
+	backupPod, err := e.createBackupPod(ctx, ownerObject, backupPVC, csiExposeParam.HostingPodLabels)
+	if err != nil {
+		return errors.Wrap(err, "error to create backup pod")
+	}
+
+	curLog.WithField("pod name", backupPod.Name).Info("Backup pod is created")
+
+	defer func() {
+		if err != nil {
+			kube.DeletePodIfAny(ctx, e.kubeClient.CoreV1(), backupPod.Name, backupPod.Namespace, curLog)
+		}
+	}()
+
+	return nil
+}
+
+func (e *csiSnapshotExposer) GetExposed(ctx context.Context, ownerObject *unstructured.Unstructured, timeout time.Duration, param interface{}) (*ExposeResult, error) {
+	exposeWaitParam := param.(*CSISnapshotExposeWaitParam)
+
+	backupPodName := ownerObject.GetName()
+	backupPVCName := ownerObject.GetName()
+
+	curLog := e.log.WithFields(logrus.Fields{
+		"owner": ownerObject.GetName(),
+	})
+
+	pod := &corev1.Pod{}
+	err := exposeWaitParam.NodeClient.Get(ctx, types.NamespacedName{
+		Namespace: ownerObject.GetNamespace(),
+		Name:      backupPodName,
+	}, pod)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			curLog.WithField("backup pod", backupPodName).Errorf("Backup pod is not running in the current node %s", exposeWaitParam.NodeName)
+			return nil, nil
+		} else {
+			return nil, errors.Wrapf(err, "error to get backup pod %s", backupPodName)
+		}
+	}
+
+	curLog.WithField("pod", pod.Name).Infof("Backup pod is in running state in node %s", pod.Spec.NodeName)
+
+	_, err = kube.WaitPVCBound(ctx, e.kubeClient.CoreV1(), e.kubeClient.CoreV1(), backupPVCName, ownerObject.GetNamespace(), timeout)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error to wait backup PVC bound, %s", backupPVCName)
+	}
+
+	curLog.WithField("backup pvc", backupPVCName).Info("Backup PVC is bound")
+
+	return &ExposeResult{ByPod: ExposeByPod{HostingPod: pod, PVC: backupPVCName}}, nil
+}
+
+func (e *csiSnapshotExposer) CleanUp(ctx context.Context, ownerObject *unstructured.Unstructured, vsName string, sourceNamespace string) {
+	backupPodName := ownerObject.GetName()
+	backupPVCName := ownerObject.GetName()
+	backupVSName := ownerObject.GetName()
+
+	kube.DeletePodIfAny(ctx, e.kubeClient.CoreV1(), backupPodName, ownerObject.GetNamespace(), e.log)
+	kube.DeletePVCIfAny(ctx, e.kubeClient.CoreV1(), backupPVCName, ownerObject.GetNamespace(), e.log)
+	csi.DeleteVolumeSnapshotIfAny(ctx, e.csiSnapshotClient, backupVSName, ownerObject.GetNamespace(), e.log)
+	csi.DeleteVolumeSnapshotIfAny(ctx, e.csiSnapshotClient, vsName, sourceNamespace, e.log)
+}
+
+func getVolumeModeByAccessMode(accessMode string) (corev1.PersistentVolumeMode, error) {
+	if accessMode == AccessModeFileSystem {
+		return corev1.PersistentVolumeFilesystem, nil
+	} else {
+		return "", errors.Errorf("unsupported access mode %s", accessMode)
+	}
+}
+
+func (e *csiSnapshotExposer) createBackupVS(ctx context.Context, ownerObject *unstructured.Unstructured, snapshotVS *snapshotv1api.VolumeSnapshot) (*snapshotv1api.VolumeSnapshot, error) {
+	backupVSName := ownerObject.GetName()
```

Comment:
It seems the VSC name using the `ownerObject` is not right, because VSC is not a namespaced-scope resource.
If two `ownerObject` from different namespaces have the same name, this will cause conflict.